internal/overlord/logstate/syslog_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package logstate
+
+import (
+	"bufio"
+	"bytes"
+	"net"
+	"net/url"
+	"time"
+
+	"github.com/canonical/pebble/internal/servicelog"
+	. "gopkg.in/check.v1"
+)
+
+type syslogSuite struct{}
+
+var _ = Suite(&syslogSuite{})
+
+func (s *syslogSuite) TestEncodeEntry(c *C) {
+	client := &syslogClient{
+		version:        1,
+		pid:            "100146",
+		host:           "mycontainer",
+		msgid:          "-",
+		priority:       priorityVal(FacilityUserLevelMessage, SeverityInformational),
+		structuredData: syslogStructuredData("pebble", canonicalPrivEnterpriseNum, nil),
+	}
+	buf := &bytes.Buffer{}
+	client.encodeEntry(buf, servicelog.Entry{
+		Time:    time.Date(2021, 5, 26, 12, 37, 1, 0, time.UTC),
+		Service: "foo",
+		Message: "this is a log entry",
+	})
+	c.Check(buf.String(), Equals, "73 <14>1 2021-05-26T12:37:01Z mycontainer foo 100146 - - this is a log entry")
+}
+
+func (s *syslogSuite) TestSyslogClient(c *C) {
+	// Start a fake syslog server
+	ln, err := net.Listen("tcp", "localhost:0") // will select unused port
+	c.Assert(err, IsNil)
+	defer ln.Close()
+
+	messages := make(chan string)
+	srv := &testSyslogServer{
+		listener: ln,
+		messages: messages,
+	}
+
+	go func() {
+		err := srv.run()
+		c.Assert(err, IsNil)
+	}()
+
+	// Create a syslog client
+	serverAddr := ln.Addr().String()
+	u, err := url.Parse("tcp://" + serverAddr)
+	c.Assert(err, IsNil)
+	client := &syslogClient{
+		version:        1,
+		pid:            "100146",
+		host:           "mycontainer",
+		msgid:          "-",
+		priority:       priorityVal(FacilityUserLevelMessage, SeverityInformational),
+		structuredData: syslogStructuredData("pebble", canonicalPrivEnterpriseNum, nil),
+		address:        u,
+	}
+
+	// Send a message to the syslog server
+	entries := []servicelog.Entry{{
+		Time:    time.Date(2023, 1, 31, 1, 23, 45, 67890, time.UTC),
+		Service: "foobar",
+		Message: "test message",
+	}}
```

Comment:
IMO we should send a couple of different messages (maybe three like you did with Loki tests) to tests different timestamps/messages.